devicetypes/stelpro/stelpro-maestro-thermostat.src/stelpro-maestro-thermostat.groovy

```diff
@@ -0,0 +1,528 @@
+	if (preciseDegrees != null) {
+		def temperatureScale = getTemperatureScale()
+		float minSetpoint = thermostatSetpointRange[minSetpointIndex]
+		float maxSetpoint = thermostatSetpointRange[maxSetpointIndex]
+
+		if (preciseDegrees >= minSetpoint && preciseDegrees <= maxSetpoint) {
+			def degrees = new BigDecimal(preciseDegrees).setScale(1, BigDecimal.ROUND_HALF_UP)
+			def celsius = (getTemperatureScale() == "C") ? degrees : (fahrenheitToCelsius(degrees) as Float).round(2)
+
+			log.debug "setHeatingSetpoint({$degrees} ${temperatureScale})"
+
+			delayBetween([
+					zigbee.writeAttribute(THERMOSTAT_CLUSTER, ATTRIBUTE_HEAT_SETPOINT, DataType.INT16, zigbee.convertToHexString(celsius * 100, 4)),
+					zigbee.readAttribute(THERMOSTAT_CLUSTER, ATTRIBUTE_HEAT_SETPOINT),
+					zigbee.readAttribute(THERMOSTAT_CLUSTER, ATTRIBUTE_PI_HEATING_STATE)
+				], 100)
+		} else {
+			log.debug "heatingSetpoint $preciseDegrees out of range! (supported: $minSetpoint - $maxSetpoint ${getTemperatureScale()})"
+		}
+	}
+}
+
+// Maintain backward compatibility with self published versions of the "Stelpro Get Remote Temperature" SmartApp
+def quickSetOutTemp(outsideTemp) {
+	setOutdoorTemperature(outsideTemp)
+}
+
+def setOutdoorTemperature(outsideTemp) {
+	def degrees = outsideTemp as Double
+	Integer tempToSend
+	def celsius = (getTemperatureScale() == "C") ? degrees : (fahrenheitToCelsius(degrees) as Float).round(2)
+
+	if (celsius < 0) {
+		tempToSend = (celsius*100) + 65536
+	} else {
+		tempToSend = (celsius*100)
+	}
+	// The thermostat expects the byte order to be a little different than we send usually
+	zigbee.writeAttribute(THERMOSTAT_CLUSTER, ATTRIBUTE_MFR_SPEC_OUT_TEMP, DataType.INT16, zigbee.swapEndianHex(zigbee.convertToHexString(tempToSend, 4)), ["mfgCode": "0x1185"])
+}
+
+def increaseHeatSetpoint() {
+	def currentMode = device.currentState("thermostatMode")?.value
+	if (currentMode != "off") {
+		float currentSetpoint = device.currentValue("heatingSetpoint")
+
+		currentSetpoint = currentSetpoint + setpointStep
+		setHeatingSetpoint(currentSetpoint)
+	}
+}
+
+def decreaseHeatSetpoint() {
+	def currentMode = device.currentState("thermostatMode")?.value
+	if (currentMode != "off") {
+		float currentSetpoint = device.currentValue("heatingSetpoint")
+
+		currentSetpoint = currentSetpoint - setpointStep
+		setHeatingSetpoint(currentSetpoint)
+	}
+}
+
+def setThermostatMode(value) {
+	log.debug "setThermostatMode($value)"
+	// Thermostat only supports heat
+}
+
+def heat() {
+	log.debug "heat"
+	// Thermostat only supports heat
+	//sendEvent("name":"thermostatMode", "value":"heat")
+}
+
+def configure() {
+	def requests = []
+	log.debug "binding to Thermostat cluster"
+
+	unschedule(scheduledUpdateWeather)
+	if (settings.zipcode) {
+		requests += updateWeather()
+		runEvery1Hour(scheduledUpdateWeather)
+	}
+
+	// This thermostat only supports heat
+	sendEvent("name":"thermostatMode", "value":"heat")
+
+	requests += delayBetween([
```

Comment:
You don't need to use `delayBetween` with any of the `zigbee.*` commands because they automatically add a delay